java/vector/src/main/java/org/apache/arrow/vector/BigIntVector.java

```diff
@@ -71,7 +73,11 @@ public BigIntVector(String name, FieldType fieldType, BufferAllocator allocator)
    */
   public BigIntVector(Field field, BufferAllocator allocator) {
     super(field, allocator, TYPE_WIDTH);
-    reader = new BigIntReaderImpl(BigIntVector.this);
+    reader = () -> {
```

Comment:
This is not thread safe, but I am not sure if adding the Google core libraries dependency to get `Suppliers::memoize` is allowed. I don't know of other ways to make this thread safe without adding any `throws` clause.

Reply:
is it possible to liberate the necessary code from either of these libraries? I'm not sure how much baggage they would drag with them

Reply:
Thanks for working on this rtadepalli. Instead of the above logic, should we add a new constructor with a new boolean parameter createReader. We create the reader based on the boolean flag. And in the getReader if the reader is null, we throw an exception. Callers who dont need the reader can call the new constructor, and if they try accessing the reader then they get an exception.